software/base/src/test/java/org/apache/brooklyn/entity/EC2ExpungeLiveTest.java

```diff
@@ -0,0 +1,128 @@
+import org.jclouds.compute.domain.NodeMetadata;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import javax.annotation.Nullable;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+public class EC2ExpungeLiveTest extends BrooklynAppLiveTestSupport {
+    private static final Logger LOG = LoggerFactory.getLogger(EC2ExpungeLiveTest.class);
+
+    public static final String PROVIDER = "aws-ec2";
+    public static final String SMALL_HARDWARE_ID = "m1.small";
+
+    public static final String REGION_NAME = "us-west-2";
+    public static final String LOCATION_SPEC = PROVIDER + (REGION_NAME == null ? "" : ":" + REGION_NAME);
+
+    protected BrooklynProperties brooklynProperties;
+
+    protected Location jcloudsLocation;
+
+    @BeforeMethod(alwaysRun=true)
+    @Override
+    public void setUp() throws Exception {
+        // Don't let any defaults from brooklyn.properties (except credentials) interfere with test
+        brooklynProperties = BrooklynProperties.Factory.newDefault();
+
+        // Also removes scriptHeader (e.g. if doing `. ~/.bashrc` and `. ~/.profile`, then that can cause "stdin: is not a tty")
+        brooklynProperties.remove("brooklyn.ssh.config.scriptHeader");
+
+        mgmt = new LocalManagementContextForTests(brooklynProperties);
+        super.setUp();
+    }
+
+    /**
+     * Verifies the behavior described in
+     * <a href="https://issues.apache.org/jira/browse/BROOKLYN-264">BROOKLYN-264 Stop app while VM still being provisioned: vm is left running when app is expunged</a>
+     * <ul>
+     *     <li>ApplicationResource.launch</li>
+     *     <li>wait a few seconds and EntityResponse.expunge</li>
+     *     <li>assert the image is on the cloud</li>
+     * </ul>
+     */
+    @Test(groups = {"Live"})
+    public void testExpunge() {
+        Map<String,String> flags = ImmutableMap.of("imageId", "us-west-2/ami-cd715dfd", LocationConfigKeys.CLOUD_MACHINE_NAMER_CLASS.getName(), "");
+        Map<String,?> allFlags = MutableMap.<String,Object>builder()
+                .put("tags", ImmutableList.of(getClass().getName()))
+                .putAll(flags)
+                .build();
+        JcloudsLocation jcloudsLocation = (JcloudsLocation)mgmt.getLocationRegistry().getLocationManaged(LOCATION_SPEC, allFlags);
+
+
+        final EmptySoftwareProcess emptySoftwareProcess = app.createAndManageChild(EntitySpec.create(EmptySoftwareProcess.class)
+                .configure(EmptySoftwareProcess.PROVISIONING_PROPERTIES.subKey(CloudLocationConfig.INBOUND_PORTS.getName()), ImmutableList.of(22)));
+
+        app.addLocations(ImmutableList.of(jcloudsLocation));
+
+        EntityManagementUtils.start(app);
+
+        try {
+            // Number of seconds we wait for an AWS EC2 instance to be created.
+            // The test succeeds if stop effector is called 200 seconds or more later
+            Thread.sleep(16 * 1000);
```

Comment:
This kind of timing feels brittle to include in a test. But I guess we do need to wait some length of time, so it's not terminating it while it was still constructing the template (rather than in the middle of actually provisioning).